hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemAuthorization.java

```diff
@@ -65,6 +66,21 @@ public void setup() throws Exception {
     super.setup();
   }
 
+  @Test
+  public void testSASTokenProviderInitializeException() throws Exception {
+    final AzureBlobFileSystem fs = this.getFileSystem();
+
+    final AzureBlobFileSystem testFs = new AzureBlobFileSystem();
+
+    MockSASTokenProvider.setThrowExceptionAtInit(true);
```

Comment:
I think this will fail intermittently when the tests are run in parallel, which is how we usually run the tests.

Reply:
I am seeing failures when running the tests

  **mvn -T 1C -Dparallel-tests=abfs -Dscale -DtestsThreadCount=8 clean verify**

[ERROR] Errors:
[ERROR]   ITestAbfsIdentityTransformer.<init>:66 » NullPointer
[ERROR]   ITestAbfsIdentityTransformer.<init>:66 » NullPointer
[ERROR]   ITestAbfsIdentityTransformer.<init>:66 » NullPointer
[ERROR]   ITestAbfsIdentityTransformer.<init>:66 » NullPointer
[ERROR]   ITestAbfsIdentityTransformer.<init>:66 » NullPointer
[ERROR]   ITestAbfsIdentityTransformer.<init>:66 » NullPointer
[ERROR]   ITestAzureBlobFileSystemAuthorization.testRenameAuthorized:125->runTest:241-
[ERROR] Tests run: 409, Failures: 0, Errors: 7, Skipped: 32

Reply:
Updated PR to fix test failures. Re-ran tests and all passed.